validator/sawtooth_validator/execution/predecessor_tree.py

```diff
@@ -86,12 +86,17 @@ def get_nodes_above(self, address):
             node = node.children[token]
             yield node
 
+
+class PredecessorTree:
+    def __init__(self, token_size=2):
+        self._tree = Tree(token_size)
+
     def add_reader(self, address, reader):
-        node = self.insert(address)
+        node = self._tree.insert(address)
```

Comment:
The radix tree should not allocate the stored data structure; instead, this should be "self._tree.insert(address, node_data)" where node could be a PredecessorTreeNode or any other python data structure.

Reply:
fixed